frontend/src/components/core/Block/ElementNodeRenderer.tsx

```diff
@@ -0,0 +1,588 @@
+        <NumberInput
+          key={numberInputProto.id}
+          element={numberInputProto}
+          width={width}
+          {...widgetProps}
+        />
+      )
+    }
+
+    case "radio": {
+      const radioProto = node.element.radio as RadioProto
+      return (
+        <Radio
+          key={radioProto.id}
+          element={radioProto}
+          width={width}
+          {...widgetProps}
+        />
+      )
+    }
+
+    case "selectbox": {
+      const selectboxProto = node.element.selectbox as SelectboxProto
+      return (
+        <Selectbox
+          key={selectboxProto.id}
+          element={selectboxProto}
+          width={width}
+          {...widgetProps}
+        />
+      )
+    }
+
+    case "slider": {
+      const sliderProto = node.element.slider as SliderProto
+      return (
+        <Slider
+          key={sliderProto.id}
+          element={sliderProto}
+          width={width}
+          {...widgetProps}
+        />
+      )
+    }
+
+    case "textArea": {
+      const textAreaProto = node.element.textArea as TextAreaProto
+      return (
+        <TextArea
+          key={textAreaProto.id}
+          element={textAreaProto}
+          width={width}
+          {...widgetProps}
+        />
+      )
+    }
+
+    case "textInput": {
+      const textInputProto = node.element.textInput as TextInputProto
+      return (
+        <TextInput
+          key={textInputProto.id}
+          element={textInputProto}
+          width={width}
+          {...widgetProps}
+        />
+      )
+    }
+
+    case "timeInput": {
+      const timeInputProto = node.element.timeInput as TimeInputProto
+      return (
+        <TimeInput
+          key={timeInputProto.id}
+          element={timeInputProto}
+          width={width}
+          {...widgetProps}
+        />
+      )
+    }
+
+    default:
+      throw new Error(`Unrecognized Element type ${node.element.type}`)
+  }
+}
+
+const ElementNodeRenderer = (
+  props: ElementNodeRendererProps
+): ReactElement => {
+  const { node } = props
+
+  const elementType = node.element.type
+  const isHidden = elementType === "empty" || elementType === "balloons"
+  const enable = shouldComponentBeEnabled(isHidden, props.reportRunState)
+  const isStale = isComponentStale(
+    enable,
+    node,
+    props.showStaleElementIndicator,
+    props.reportRunState,
+    props.reportId
+  )
+
+  // TODO: Move this into type signature of props. The width is actually guaranteed to be nonzero
+  // since leaf elements are always direct children of a VerticalBlock, which always calculates
+  // and propagates widths.
+  const width = props.width ?? 0
+
+  // TODO: If would be great if we could return an empty fragment if isHidden is true, to keep the
+  // DOM clean. But this would require the keys passed to ElementNodeRenderer at Block.tsx to be a
+  // stable hash of some sort.
+
+  return (
+    <Maybe enable={enable}>
+      <StyledElementContainer
+        data-stale={!!isStale}
```

Comment:
I don't think this is necessary, since `isStale` comes from a function that definitely returns a bool. Simplify to just `data-stale=isStale`?

Reply:
Done. BTW, this came from the original code. I was trying to keep changes to a minimum.

https://github.com/streamlit/streamlit/blob/develop/frontend/src/components/core/Block/Block.tsx#L317